content/guides/end-to-end-testing/testing-your-app.md

```diff
@@ -410,111 +410,87 @@ You'll likely also want to test your login UI for:
 
 Each of these likely requires a full blown e2e test.
 
-Now, once you have your login completely tested - you may be tempted to think:
+#### Reusing the login code
 
-> "...okay great! Let's repeat this login process before every single test!"
+At this point there's nothing stopping you copying and pasting the login code
+above into every one of your tests that needs an authenticated user. Or you
+could even put all your tests in one big spec file and put the login code in a
+`beforeEach` block. But neither of those approaches is particularly
+maintainable, and they're certainly not very elegant. A much better solution is
+to use a `cy.login()` [custom command](/api/cypress-api/custom-commands).
 
-<Alert type="danger">
+Custom commands allow you to easily encapsulate and reuse Cypress test logic.
+They're basically just functions, but with some added sugar to allow you to use
+them with the same chaining API as the built-in Cypress commands. Here's an
+example of the login code from the previous section converted into a custom
+command:
 
```

Comment:
```suggestion
Lets make the above login example a custom command, we added to `cypress/support/commands.js` so it can be leveraged in any spec file:
```